### PR TITLE
feat: Release version support for cache generation

### PR DESCRIPTION
**Pull request by ivadym**
### This PR:

# Added:
- Allow cache download specifying the future release version (before bumping the Balsamic package version).

### Review and tests: 
- [x] Tests pass

  - [x] `balsamic init` (current balsamic version) Screenshot 2023-08-23 at 14 48 59
 
  - [x] `balsamic init --cache-version develop` Screenshot 2023-08-23 at 14 50 48Screenshot 2023-08-23 at 14 59 20

  - [x] `balsamic init --cache-version 13.0.0` Screenshot 2023-08-23 at 14 52 20 Screenshot 2023-08-23 at 14 53 47

- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Review by mathiasbio — APPROVED** [review]
Sooo to see if I understood this correctly I'll try to repeat how this will be used in our development and deployment workflow:
- We're working on develop, and if there's any changes to the containers they will be published on dockerhub and with this update we can easily pull it down to keep our PRs up to date by just building the develop cache: `balsamic init --cache-version develop` 
When it's time to make the release from develop:
- We make sure master is up to date and develop is up to date, and merge master into develop just in case there's been any fixes we haven't added to develop.
- We make a new release branch from develop, and we give it a name fitting the format release_vX.X.X and update the CHANGELOG version number to the same X.X.X. 
- At this point after the release branch has been made, github actions will publish the containers to dockerhub, and these are the containers we will use for verification, validation, and if all goes well, production, and with this update we can prepare this release cache for verification and validation with` balsamic init --cache-version 13.0.0`

Two questions:
1. Do we still copy over the cache that we built for the verification, or do we pull down again with the init command? I suppose since the containers are identical, we should pull them down again, to avoid needing to manually change the paths in the reference json etc! 
2. If something goes wrong, we go back to develop, make the fixes, create a new release branch which will push the updated containers and container-versions? Or what is the flow? 

In conclusion! It looks great and clean! And it's really nice to have this streamlined workflow for deployments :))) 🌟  ⭐  🌟

**Comment by ivadym** [comment]
> 1. Do we still copy over the cache that we built for the verification, or do we pull down again with the init command? I suppose since the containers are identical, we should pull them down again, to avoid needing to manually change the paths in the reference json etc!

Exactly, this PR is just to pull a different version from the current balsamic one. I'll be removing the absolute paths from the references in the future so we can actually copy paste it from stage to production.

> 2. If something goes wrong, we go back to develop, make the fixes, create a new release branch which will push the updated containers and container-versions? Or what is the flow?

Exactly, we'll have to go through a release branch every time we make a new patch/minor/major release.


Thanks for the review @mathiasbio 💟
